doc/manual/source/store/drv.md

```diff
@@ -0,0 +1,284 @@
+> we ought to consider switching two a different format where we explicitly use a syntax for a oncatentation of plain strings and placeholders written more explicitly.
+
+## Deriving path {#deriving-path}
+
+Deriving paths are a way to refer to [store objects][store object] that might not yet be [realised][realise].
+This is necessary because, in general and particularly for [content-addressed derivations][content-addressed derivation], the [store path] of an [output] is not known in advance.
+There are two forms:
+
+- *constant*: just a [store path]
+  It can be made [valid][validity] by copying it into the store: from the evaluator, command line interface or another st    ore.
+
+- *output*: a pair of a [store path] to a [derivation] and an [output] name.
+
+In pseudo code:
+
+```idris
+type OutputName = String
+
+data DerivingPath
+  = ConstantPath { path : StorePath }
+  | Output {
+      drvPath : StorePath,
+      output  : OutputName,
+    }
+```
+
+[deriving path]: #deriving-path
+
+[content-addressed derivation]: @docroot@/glossary.md#gloss-content-addressed-derivation
+[realise]: @docroot@/glossary.md#gloss-realise
+[store object]: @docroot@/store/store-object.md
+[store path]: @docroot@/store/store-path.md
+
+## Encoding
+
+### Derivation {#derivation-encoding}
+
+There are two formats, documented separately:
+
+- The legacy ["ATerm" format](@docroot@/protocols/derivation-aterm.md)
+
+- The modern [JSON format](@docroot@/protocols/json/derivation.md)
+
+Currently derivations are always serialized to store objects using the "ATerm" format, but this is subject to change.
+
+Regardless of the format used, when serializing to store object, content-addressing is always used.
+In the common case the inputs to store objects are either:
+
+ - constant deriving paths for content-addressed source objects, which are "initial inputs" rather than the outputs of some other derivation (except in the case of bootstrap binaries).
+
+ - the outputs of other derivations abiding by this same invariant.
+
+This common case makes for the following useful property:
+when we serialize such a derivation graph to store objects, the resulting closures are *entirely* content-addressed.
+
+Here is a sketch at the proof of this:
+
+ - The inputs which are constant deriving paths become references of the serialized derivations, but they are content-addressed per the above.
+
+ - For inputs which are output deriving paths, we cannot directly reference the input because in general it is not built yet.
+   We instead "peal back" the output deriving path to take its underlying serialized derivation (the `drvPath` field), and reference that.
+   Since it is a derivation, it must be content-addressed
+
+ - There are no other ways a store object would end up in an input closure.
+   The references of a derivation in store object form always come from solely from the inputs of the derivation.
```

Comment:
Except for `builtins.storePath`, `builtins.appendContext`, as well as `builtins.fetchClosure` with certain parameters.

Reply:
Hmm? Those will all create `inputSrcs`.

It is true that those can be used to create non-content-addressed `inputSrcs`, but that falls outside of the common case mentioned above.